frontend/components/product-list/sections/FilterableProductsSection/ProductListFilters/FilterRow.tsx

```diff
@@ -309,3 +319,25 @@ function avg(x: string): number | null {
    }
    return nums.reduce((x, y) => x + parseFloat(y), 0) / nums.length;
 }
+
+function compareCapacity(a: FacetOption, b: FacetOption): number {
+   return capacityStringToBytes(b.value) - capacityStringToBytes(a.value);
+}
+
+const unitToBytes = {
+   'B': 1,
+   'KB': 1024,
+   'MB': 1024 * 1024,
+   'GB': 1024 * 1024 * 1024,
+   'TB': 1024 * 1024 * 1024 * 1024,
+   'PB': 1024 * 1024 * 1024 * 1024 * 1024,
+};
+
+function capacityStringToBytes(x: string): number {
+   const size = parseFloat(x);
+   if (isNaN(size)) {
+      return 0;
+   }
+   const unit = x.replace(/[0-9.]+/, '').trim() as keyof typeof unitToBytes;
```

Comment:
```suggestion
   const unit = x.match(/.B/)?.[0] ?? 'B' as keyof typeof unitToBytes;
```

No idea if that's faster, but I understand it better.

Reply:
Implemented something similar: https://github.com/iFixit/react-commerce/pull/277/commits/65d02173a3be9e8f5e7bb0f49ea3f78adc2a2980